### DISCUSSION TITLE
How to add a more planet, e.g "Ceres" in astronomy.js or astronomy.ts?

### DISCUSSION THREAD

**Discussion by testori1972** [cosinekitty/astronomy#351] [General]
How to add a more planet?

**Comment by cosinekitty** [comment]
This is not simple, nor is it a small task.

First, find a reliable source of correct data for the planet, such as an ephemeris or another authoritative algorithm. Astronomy Engine uses VSOP87 for the major planets, which includes Mercury, Venus, Earth, Mars, Jupiter, Saturn, Uranus, and Neptune.

Once you find a model, figure out how to truncate it as needed to fit inside a small code footprint (especially important for JavaScript support in browsers) without excessive angular error (1 arcminute max, as seen from Earth, is our target). Write unit tests that verify the algorithm against the authoritative test data/code.

Then implement it for all supported programming languages: C/C++, C#, TypeScript/JavaScript, Python, and Kotlin. Create unit tests, documentation, and demos. Verify all work.

Currently this is not planned unless someone else wants to volunteer to do all this work. It would be a valuable contribution for sure! If so, it would be better to go ahead and implement the four major asteroids together: Ceres, Pallas, Vesta, and Juno.

**Reply by testori1972** [reply]
I just updated the script called: astronomy.ts by adding, eclipse magnitude, equatorial rotation velocity, central eclipse, along with distance, azimuth, equatorial velocity which has been patched with the same function, namely PlanetVelocity, unfortunately width of umbra, etc. is still being worked on

**Reply by testori1972** [reply]
> This is not simple, nor is it a small task.
> 
> First, find a reliable source of correct data for the planet, such as an ephemeris or another authoritative algorithm. Astronomy Engine uses VSOP87 for the major planets, which includes Mercury, Venus, Earth, Mars, Jupiter, Saturn, Uranus, and Neptune.
> 
> Once you find a model, figure out how to truncate it as needed to fit inside a small code footprint (especially important for JavaScript support in browsers) without excessive angular error (1 arcminute max, as seen from Earth, is our target). Write unit tests that verify the algorithm against the authoritative test data/code.
> 
> Then implement it for all supported programming languages: C/C++, C#, TypeScript/JavaScript, Python, and Kotlin. Create unit tests, documentation, and demos. Verify all work.
> 
> Currently this is not planned unless someone else wants to volunteer to do all this work. It would be a valuable contribution for sure! If so, it would be better to go ahead and implement the four major asteroids together: Ceres, Pallas, Vesta, and Juno.

I'm looking for a scheme to calculate the planet Pluto, it's very complicated so I have to do console.log() starting from booleans, numbers, gravity, etc. I even had to search in the "/generate" folder to find the same value, not only that, I searched in the "/generate" folder, and the related site in the "PlutoStateTable" variable, then I would compress it into javascript code to add the name of the example planet "Ceres" @cosinekitty
